irods_testing_environment/irods_setup.py

```diff
@@ -264,8 +264,19 @@ def build_input_for_catalog_consumer(self):
             '' # confirmation of inputs
         ]
 
+        # Handle the difference between 4.3 servers and 5.x servers.
+        if self.irods_version >= (5, 0, 0):
+            # Insert entries for iRODS 4.3.
+            input_args.insert(3, str(self.provides_local_storage))
+            input_args.insert(4, str(self.resource_name))
+            input_args.insert(5, str(self.vault_directory))
+
+            # Remove entries that do not apply to iRODS 5.
+            del input_args[11] # Control plane port
+            del input_args[12] # Schema validation base URI
+            del input_args[17] # Control plane key
```

Comment:
This is incorrect. As soon as element 11 is removed, the indices become invalid.

The better solution is to remove by matching on values - i.e. `.remove(element)`